File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/navigation/NavigationControllerFactory.java
package com.github.willjgriff.ethereumwallet.ui.navigation;

import android.view.MenuItem;

import com.bluelinelabs.conductor.Controller;
import com.github.willjgriff.ethereumwallet.ui.nodestatus.NodeStatusController;
import com.github.willjgriff.ethereumwallet.ui.transactions.TransactionsController;
import com.github.willjgriff.ethereumwallet.R;
import com.github.willjgriff.ethereumwallet.ui.receive.ReceiveController;
import com.github.willjgriff.ethereumwallet.ui.send.SendController;
import com.github.willjgriff.ethereumwallet.ui.settings.SettingsController;

/**
 * Created by Will on 29/01/2017.
 */

public class NavigationControllerFactory {

	public Controller getController(MenuItem item) {
		switch (item.getItemId()) {
			case R.id.navigation_transactions:
				return new TransactionsController();
			case R.id.navigation_send:
				return new SendController();
			case R.id.navigation_receive:
				return new ReceiveController();
			case R.id.navigation_status:
				return new NodeStatusController();
			case R.id.navigation_settings:
				return new SettingsController();
		}
		return null;
	}
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/navigation/NavigationControllerFactory.kt
package com.github.willjgriff.ethereumwallet.ui.navigation

import android.view.MenuItem
import com.bluelinelabs.conductor.Controller
import com.github.willjgriff.ethereumwallet.R
import com.github.willjgriff.ethereumwallet.ui.screens.nodedetails.NodeDetailsController
import com.github.willjgriff.ethereumwallet.ui.screens.receive.ReceiveController
import com.github.willjgriff.ethereumwallet.ui.screens.send.SendController
import com.github.willjgriff.ethereumwallet.ui.screens.settings.SettingsController
import com.github.willjgriff.ethereumwallet.ui.screens.transactions.TransactionsController

/**
 * Created by williamgriffiths on 04/05/2017.
 */
class NavigationControllerFactory {

    fun getController(item: MenuItem): Controller? {
        when (item.itemId) {
            R.id.navigation_transactions -> return TransactionsController()
            R.id.navigation_send -> return SendController()
            R.id.navigation_receive -> return ReceiveController()
            R.id.navigation_status -> return NodeDetailsController()
            R.id.navigation_settings -> return SettingsController()
            else -> return null
        }
    }
}